py/pysparkling/context.py

```diff
@@ -156,10 +155,17 @@ def as_spark_frame(self, h2o_frame, copy_metadata=True):
         -------
           Spark DataFrame
         """
-        if isinstance(fr, H2OFrame):
-            j_h2o_frame = fr.get_java_h2o_frame()
+        if isinstance(h2o_frame, H2OFrame):
+            j_h2o_frame = h2o_frame.get_java_h2o_frame()
             jdf = self._jhc.asDataFrame(j_h2o_frame, copy_metadata, self._jsql_context)
-            return DataFrame(jdf, self._sql_context)
+            df = DataFrame(jdf, self._sql_context)
+            # Attach h2o_frame to dataframe which forces python not to delete the frame when we leave the scope of this
+            # method.
+            # Without this, after leaving this method python would garbage collect the frame since it's not used
+            # anywhere and spark. when executing any action on this dataframe, will fail since the frame
+            # would be missing.
+            df._h2o_frame = h2o_frame
```

Comment:
One note: in theory we should redefine `_del_` method on `DataFrame` to kill also `_h2o_frame` reference. But for now, it is good enough to unblock cust.